lib/spack/spack/test/data/compilers.yaml

```diff
@@ -8,6 +8,7 @@ compilers:
       f77: None
       fc: None
     modules: 'None'
+    implicit_rpaths: []
```

Comment:
Is this required now?  If so we should avoid that, as it'll break existing configs that do not have this field.  We should also do some separate PR that makes the YAML for this stuff more minimal by omitting empty entries.

Reply:
This PR does not brack existing configs that do not have this field. Confirmed on my mac.

Reply:
ok good.  I don't think we need to add this here then do we?

Reply:
e.g., we already do not have an `extra_rpaths` field -- why change it?